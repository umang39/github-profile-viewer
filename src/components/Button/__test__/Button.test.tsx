import React from "react";
import { cleanup, fireEvent, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { ButtonInterface, ButtonComponent } from "@components/index";
afterEach(cleanup);
const handleClick = jest.fn();
const buttonProps: ButtonInterface = {
  btnHandler: handleClick,
  value: "submit",
  type: "button",
  id: "login-btn",
  isDisabled: false,
};
const buttonComponentHandler = (props: ButtonInterface = buttonProps) => {
  return render(
    <ButtonComponent
      btnHandler={props.btnHandler}
      value={props.value}
      type={props.type}
      id={props.id}
      isDisabled={props.isDisabled}
    />
  );
};
const snapButtonHandler = (props: ButtonInterface = buttonProps) => {
  return renderer
    .create(
      <ButtonComponent
        btnHandler={props.btnHandler}
        value={props.value}
        type={props.type}
        id={props.id}
        isDisabled={props.isDisabled}
      />
    )
    .toJSON();
};
it("button running without crashing", () => {
  const { container } = buttonComponentHandler();
  expect(container).toBeInTheDocument();
});
it("button is not disbabled", () => {
  buttonComponentHandler();
  const disbledFeature = (screen.queryByRole("button") as HTMLInputElement)
    .disabled;
  fireEvent.click(screen.queryByRole("button") as HTMLInputElement);
  expect(disbledFeature).toBe(false);
  expect(handleClick).toHaveBeenCalled();
});
it("follow button in social card running successfully", () => {
  buttonComponentHandler({
    type: "button",
    value: "Follow",
    id: "umang39",
    btnHandler: jest.fn(),
    isDisabled: false,
  });
  const disbledFeature = (screen.queryByRole("button") as HTMLInputElement)
    .disabled;
  fireEvent.click(screen.queryByRole("button") as HTMLInputElement);
  expect(disbledFeature).toBe(false);
  expect(handleClick).not.toHaveBeenCalled();
});
it("button is disbabled", () => {
  const { getByRole } = render(
    <ButtonComponent
      btnHandler={jest.fn()}
      value="submit"
      type="button"
      id="login-btn"
      isDisabled={true}
    />
  );
  fireEvent.click(screen.queryByRole("button") as HTMLInputElement);
  expect(getByRole("button")).toBeDisabled();
  expect(handleClick).not.toHaveBeenCalled();
});
it("matches snapshot with login button disabled", () => {
  const tree = snapButtonHandler({
    btnHandler: handleClick,
    value: "submit",
    type: "button",
    id: "login-btn",
    isDisabled: true,
  });
  expect(tree).toMatchSnapshot();
});
it("matches snapshot with login button not disabled", () => {
  const tree = snapButtonHandler();
  expect(tree).toMatchSnapshot();
});
it("matches snapshot with follow button", () => {
  const tree = snapButtonHandler({
    type: "button",
    value: "Follow",
    id: "umang39",
    btnHandler: jest.fn(),
    isDisabled: false,
  });
  expect(tree).toMatchSnapshot();
});
