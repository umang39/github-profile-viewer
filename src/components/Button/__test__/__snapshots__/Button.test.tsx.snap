// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with follow button 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-5"
  disabled={false}
  id="umang39"
  onBlur={[Function]}
  onClick={[MockFunction]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    Follow
  </span>
</button>
`;

exports[`matches snapshot with login button disabled 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-5 Mui-disabled Mui-disabled"
  disabled={true}
  id="login-btn"
  onBlur={[Function]}
  onClick={[MockFunction]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={-1}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    submit
  </span>
</button>
`;

exports[`matches snapshot with login button not disabled 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-5"
  disabled={false}
  id="login-btn"
  onBlur={[Function]}
  onClick={[MockFunction]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    submit
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;
